temperatureEndpoint.py

```diff
@@ -0,0 +1,62 @@
+from urllib.request import urlopen
+from urllib.error import URLError, HTTPError
+import json
+import abc
+
+class AbstractDataService(metaclass=abc.ABCMeta):
+    def get_weather_forecast(self):
+        pass
+
+# Data for testing
+class MockedDataService(AbstractDataService):
+    def get_weather_forecast(self):
+        return {"forecast": "sunny", "temp": 25}
+    
+# API Calling Service
+class APICallingService(AbstractDataService):
+    def __init__(self, api_url):
```

Comment:
you don't really need to pass the `api_url` in as a constructor parameter, since your class already expects a certain type of API to call (your `get_weather_forecast` is built for open-meteo), you can instead make it a static class member attribute.

In other words, you'd probably never use the `APICallingService` class for anything other than `open-meteo` (and the specific endpoint you're calling), so there is no need to make it "configurable"